core/domain/stats_jobs_one_off.py

```diff
@@ -1280,29 +1291,63 @@ def map(exp):
                 exp.id, version)
             exp_at_version = exp_models.ExplorationModel.get_version(
                 exp.id, version)
-            # Delete all old models.
-            if all_models[version - 1] is None:
-                if commit_log.commit_type != 'revert':
-                    yield ('Missing model without revert commit', exp.id)
-                    return
-
-                # Is a revert commit.
-                revert_to_version = commit_log.commit_cmds[0]['version_number']
+
+            if commit_log is None:
+                prev_exp = exp_models.ExplorationModel.get_version(
+                    exp.id, version - 1)
+                old_states = prev_exp.states
+                new_states = exp_at_version.states
+                inferred_change_list = []
+                for old_state in old_states:
+                    if old_state not in new_states:
+                        inferred_change_list.append(
+                            exp_domain.ExplorationChange({
+                                'cmd': exp_domain.CMD_DELETE_STATE,
+                                'state_name': old_state
+                            }))
+
+                for new_state in new_states:
+                    if new_state not in old_states:
+                        inferred_change_list.append(
+                            exp_domain.ExplorationChange({
+                                'cmd': exp_domain.CMD_ADD_STATE,
+                                'state_name': new_state
+                            }))
+                exp_versions_diff = exp_domain.ExplorationVersionsDiff(
+                    inferred_change_list)
+
+                if all_models[version - 1] is not None:
+                    all_models[version - 1].delete()
```

Comment:
Could you explain, in detail, why we are deleting this? (Maybe in a comment)

Reply:
Added comments